File: t/PACSFramework/VB/DicomHighLevelMWLScuDemo/Queries/PatientBasedQuery.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Leadtools.Dicom.Common.DataTypes
Imports Leadtools.Dicom.Common.Extensions
Imports Leadtools.Dicom
Imports System.ComponentModel
Imports Leadtools.Dicom.Common.Editing.Converters

Namespace DicomDemo.Queries
	Friend Class PatientBasedQuery
		Private _PatientName As PersonName = New PersonName()

		<TypeConverter(GetType(PersonNameConverter)), DisplayName("Person Name")> _
		Public Property PatientName() As PersonName
			Get
				Return _PatientName
			End Get
			Set
				_PatientName = Value
			End Set
		End Property

		Private _PatientId As String = String.Empty

		<DisplayName("Patient ID")> _
		Public Property PatientId() As String
			Get
				Return _PatientId
			End Get
			Set
				_PatientId = Value
			End Set
		End Property

		Private _AccessionNumber As String = String.Empty

		<DisplayName("Accession #")> _
		Public Property AccessionNumber() As String
			Get
				Return _AccessionNumber
			End Get
			Set
				_AccessionNumber = Value
			End Set
		End Property

		Private _RequestedProcedureId As String = String.Empty

		<DisplayName("Requested Procedure ID")> _
		Public Property RequestedProcedureId() As String
			Get
				Return _RequestedProcedureId
			End Get
			Set
				_RequestedProcedureId = Value
			End Set
		End Property
	End Class
End Namespace

File: t/PACSFramework/CS/DicomHighLevelMWLScuDemo/Queries/PatientBasedQuery.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.Text;
using Leadtools.Dicom.Common.DataTypes;
using Leadtools.Dicom.Common.Extensions;
using Leadtools.Dicom;
using System.ComponentModel;
using Leadtools.Dicom.Common.Editing.Converters;

namespace DicomDemo.Queries
{
    class PatientBasedQuery
    {
        private PersonName _PatientName = new PersonName();
        
        [TypeConverter(typeof(PersonNameConverter))]
        [DisplayName("Person Name")]
        public PersonName PatientName
        {
            get { return _PatientName; }
            set { _PatientName = value; }
        }

        private string _PatientId = string.Empty;

        [DisplayName("Patient ID")]
        public string PatientId
        {
            get { return _PatientId; }
            set { _PatientId = value; }
        }

        private string _AccessionNumber = string.Empty;

        [DisplayName("Accession #")]
        public string AccessionNumber
        {
            get { return _AccessionNumber; }
            set { _AccessionNumber = value; }
        }

        private string _RequestedProcedureId = string.Empty;

        [DisplayName("Requested Procedure ID")]
        public string RequestedProcedureId
        {
            get { return _RequestedProcedureId; }
            set { _RequestedProcedureId = value; }
        }
    }
}
